com.unity.multiplayer.mlapi/Tests/Runtime/Metrics/NetworkObjects/NetworkObjectMetricsTests.cs

```diff
@@ -0,0 +1,228 @@
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            var objectSpawnedSentMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            Assert.AreEqual(1, objectSpawnedSentMetricValues.Count);
+
+            var objectSpawned = objectSpawnedSentMetricValues.First();
+            Assert.AreEqual(m_Client.LocalClientId, objectSpawned.Connection.Id);
+            Assert.AreEqual(m_NewNetworkObjectName, objectSpawned.NetworkId.Name);
+        }
+
+        [UnityTest]
+        public IEnumerator TrackNetworkObjectSpawnReceivedMetric()
+        {
+            var waitForMetricEvent = new WaitForMetricValues<ObjectSpawnedEvent>(m_ClientMetrics.Dispatcher, MetricNames.ObjectSpawnedReceived);
+
+            m_NewNetworkObject.Spawn();
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            var objectSpawnedReceivedMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            Assert.AreEqual(1, objectSpawnedReceivedMetricValues.Count);
+
+            var objectSpawned = objectSpawnedReceivedMetricValues.First();
+            Assert.AreEqual(m_Server.LocalClientId, objectSpawned.Connection.Id);
+            Assert.AreEqual(m_NewNetworkObject.NetworkObjectId, objectSpawned.NetworkId.NetworkId);
+            Assert.AreEqual($"{m_NewNetworkObjectName}(Clone)", objectSpawned.NetworkId.Name);
+        }
+
+        [UnityTest]
+        public IEnumerator TrackNetworkObjectDestroySentMetric()
+        {
+            m_NewNetworkObject.Spawn();
+
+            var waitForMetricEvent = new WaitForMetricValues<ObjectDestroyedEvent>(m_ServerMetrics.Dispatcher, MetricNames.ObjectDestroyedSent);
+            // TODO: is there a better way of waiting here?
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            m_Server.SpawnManager.OnDespawnObject(m_NewNetworkObject, true);
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            var objectDestroyedSentMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            // As there's a client and server, this event is emitted twice.
+            Assert.AreEqual(2, objectDestroyedSentMetricValues.Count);
+
+            var objectDestroyed = objectDestroyedSentMetricValues.Last();
+            Assert.AreEqual(m_Client.LocalClientId, objectDestroyed.Connection.Id);
+            Assert.AreEqual(m_NewNetworkObjectName, objectDestroyed.NetworkId.Name);
+        }
+
+        [UnityTest]
+        public IEnumerator TrackNetworkObjectDestroyReceivedMetric()
+        {
+            m_NewNetworkObject.Spawn();
+            var waitForMetricEvent = new WaitForMetricValues<ObjectDestroyedEvent>(m_ClientMetrics.Dispatcher, MetricNames.ObjectDestroyedReceived);
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            m_Server.SpawnManager.OnDespawnObject(m_NewNetworkObject, true);
+            yield return waitForMetricEvent.Wait(60);
```

Comment:
If we have multiple different values for this, we should just default to the highest value, since we exit early if we find the metric values anyways.